pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,270 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	runShared "github.com/cli/cli/pkg/cmd/run/shared"
+	"github.com/cli/cli/pkg/cmd/workflow/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/markdown"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    cmdutil.Browser
+
+	Selector string
+	Ref      string
+	Web      bool
+	Prompt   bool
+	Raw      bool
+	Yaml     bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := &ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Browser:    f.Browser,
+	}
+
+	cmd := &cobra.Command{
+		Use:    "view [<run-id> | <file name>]",
+		Short:  "View the summary of a workflow",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		Example: heredoc.Doc(`
+		  # Interactively select a workflow to view
+		  $ gh workflow view
+
+		  # View a specific workflow
+		  $ gh workflow view 0451
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.Raw = !opts.IO.CanPrompt()
+
+			if len(args) > 0 {
+				opts.Selector = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow argument required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runView(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "Open workflow in the browser")
+	cmd.Flags().BoolVarP(&opts.Yaml, "yaml", "y", false, "View the workflow yaml file")
+	cmd.Flags().StringVarP(&opts.Ref, "ref", "r", "", "The branch or tag name which contains the version of the workflow file you'd like to view")
+
+	return cmd
+}
+
+func runView(opts *ViewOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	var workflow *shared.Workflow
+	states := []shared.WorkflowState{shared.Active}
+	workflow, err = shared.ResolveWorkflow(opts.IO, client, repo, opts.Prompt, opts.Selector, states)
+	if err != nil {
+		return err
+	}
+
+	if opts.Web {
+		var url string
+		hostname := repo.RepoHost()
+		if opts.Yaml {
```

Comment:
This is not something that we have done previously, as far as I could tell, the URL that gets opened changes to view the yaml file if the web flag and yaml flag are specified. Do people like this behavior?

Reply:
I think this is reasonable; curious what @ampinsk thinks.

Reply:
I also think it's reasonable.